demos/imagenet_classification_demo/main.cpp

```diff
@@ -0,0 +1,508 @@
+            const cv::Mat& tmpImage = cv::imread(name);
+            if (tmpImage.data == nullptr) {
+                std::cerr << "Could not read image " << name << '\n';
+                imageNames.erase(imageNames.begin() + i);
+                inputImagesCount--;
+                i--;
+            }
+            else {
+                inputImages.push_back(tmpImage);
+                imageNames[i] = name.substr(name.rfind('/') + 1);
+            }
+        }
+        // ---------------------------------------------------------------------------------------------------
+
+        // ----------------------------------------Read image classes-----------------------------------------
+        bool hasClasses = !FLAGS_classes.empty();
+        std::vector<std::string> classIndexes;
+        if (hasClasses) {
+            std::vector<std::pair<std::string, std::string>> classIndexesMap;
+            std::ifstream inputClassesFile(FLAGS_classes);
+            while (true) {
+                std::string imageName;
+                std::string classIndex;
+                inputClassesFile >> imageName >> classIndex;
+                if (inputClassesFile.eof()) break;
+                classIndexesMap.push_back({imageName.substr(imageName.rfind('/') + 1), classIndex});
+            }
+            for (size_t i = 0; i < inputImagesCount; i++) {
+                for (size_t j = 0; j < classIndexesMap.size(); j++) {
+                    if (imageNames[i] == classIndexesMap[j].first) {
+                        classIndexes.push_back(classIndexesMap[j].second);
+                        break;
+                    }
+                }
+                if (i+1 != classIndexes.size()) {
+                    throw std::runtime_error("No class specified for image " + imageNames[i]);
+                }
+            }
+            classIndexesMap.clear();
+        }
+        // ---------------------------------------------------------------------------------------------------
+
+        // --------------------------------------------Read labels--------------------------------------------
+        std::vector<std::string> labels;
+        std::ifstream inputLabelsFile(FLAGS_labels);
+        std::string labelsLine;
+        while (std::getline(inputLabelsFile, labelsLine)) {
+            labels.push_back(labelsLine.substr(0, labelsLine.find(',')));
+        }
+        // ---------------------------------------------------------------------------------------------------
+
+        // -------------------------------------------Read network--------------------------------------------
+        Core ie;
+        CNNNetwork network = ie.ReadNetwork(FLAGS_m);
+        // ---------------------------------------------------------------------------------------------------
+
+        // ------------------------------------------Reshape network------------------------------------------
+        auto input_shapes = network.getInputShapes();
```

Comment:
Model's interface should be verified: number of inputs and outputs, number of channels for inputs and outputs.
This would also be nice to check that size of output is the same as the number of classes in a file provided by `-labels`.

Please stick to a single stile of var naming for example camelCase.

Reply:
It should be verified that `input_shapes` size is 1.